cpu.go

```diff
@@ -1,18 +1,30 @@
 package sqrl
 
 import (
-	"fmt"
-	"log"
+	"runtime"
+	"strings"
 
 	"github.com/shirou/gopsutil/cpu"
 )
 
-func CPUInfo() error {
+type CPUInfo struct {
+	CPU   int    `json:"cpu"`
+	Cores int32  `json:"cores"`
+	Speed string `json:"speed"`
+	Model string `json:"model"`
+}
+
+func GetCPUInfo() (CPUInfo, error) {
 	info, err := cpu.Info()
 	if err != nil {
-		log.Printf("error getting cpu: %v", err)
-		return err
+		return CPUInfo{}, err
+	}
+	mInfo := strings.Split(info[0].ModelName, "@") // mInfo contains speed and model
+	c := CPUInfo{
+		CPU:   runtime.NumCPU(),
+		Cores: info[0].Cores,
```

Comment:
I think we might have a bug here.... We are only checking the Cores for the first CPU that returns... Is it possible that a CPU could have different number of cores?  Lets investigate a bit more.